### ISSUE TITLE
When the mosaic parameter is not set to 1, train error

### ISSUE THREAD

**Issue by guishilike** [ultralytics/ultralytics#386]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

```
Error executing job with overrides: []
Traceback (most recent call last):
  File "H:/github/ultralytics/ultralytics/yolo/v8/detect/train.py", line 212, in train
    model.train(**cfg)
  File "H:\github\ultralytics\ultralytics\yolo\engine\model.py", line 193, in train
    self.trainer.train()
  File "H:\github\ultralytics\ultralytics\yolo\engine\trainer.py", line 185, in train
    self._do_train(int(os.getenv("RANK", -1)), world_size)
  File "H:\github\ultralytics\ultralytics\yolo\engine\trainer.py", line 283, in _do_train
    for i, batch in pbar:
  File "D:\anaconda3\envs\pytorch\lib\site-packages\tqdm\std.py", line 1195, in __iter__
    for obj in iterable:
  File "D:\anaconda3\envs\pytorch\lib\site-packages\torch\utils\data\dataloader.py", line 530, in __next__
    data = self._next_data()
  File "D:\anaconda3\envs\pytorch\lib\site-packages\torch\utils\data\dataloader.py", line 570, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "D:\anaconda3\envs\pytorch\lib\site-packages\torch\utils\data\_utils\fetch.py", line 52, in fetch
    return self.collate_fn(data)
  File "H:\github\ultralytics\ultralytics\yolo\data\dataset.py", line 173, in collate_fn
    value = torch.stack(value, 0)
TypeError: expected Tensor as element 1 in argument 0, but got str

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```

### Minimal Reproducible Example

The key values in the same batch are different
![image](https://user-images.githubusercontent.com/51023060/212582016-ea74ad43-5643-484a-be06-498ccfbe6827.png)
![image](https://user-images.githubusercontent.com/51023060/212581806-2d6d1cc6-1e2e-4d4a-9d88-4854d71c3d02.png)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by AyushExel** [comment]
@guishilike please provide a command to reproduce

**Comment by guishilike** [comment]
python ultralytics/yolo/v8/detect/train.py  model=yolov8s.yaml data=coco128.yaml epochs=300 imgsz=640 batch=2 mosaic=0.3

**Comment by Laughing-q** [comment]
@guishilike if you're training detection and want to modify `mosaic`, probably you can add `v5loader=True`.
the new loader haven't supported modifying `mosaic` ratio, we'll work on it. :)

**Comment by Laughing-q** [comment]
@AyushExel @glenn-jocher adding TODO flag

**Comment by Laughing-q** [comment]
@guishilike hi this issue has solved in this PR #699 which probably will be merged later today. :)

**Comment by glenn-jocher** [comment]
Removing TODO as issue is now resolved, i.e. for `mosaic=0.5`:

![train_batch0](https://github.com/ultralytics/ultralytics/assets/26833433/183bae00-7ad8-4777-bf26-a6071bdc3ec5)
